### ISSUE TITLE
Array forEach is not the function you're looking for

### ISSUE THREAD

**Issue by mflorence99** [Alex-Rafter/pragprog-exercises#5]
Consider your loop:

```js
arr1.forEach((item1) =>
  arr2.forEach((item2) => {
    if (item1 !== item2) {
      return false;
    }
  })
);
```

All the return does is exit the function that the inner loop runs, which was going to happen anyway. It doesn't terminate either the inner loop or the outer loop. Now that can be done with `for` loops and `break` statements, but that would be a crappy solution because of the O(n^2) complexity.

One thing to remember is that `forEach` is the weakest of the array functions. There's almost always a better one, like `map` or in this case `every`.

```js
return p.every((letter, ix) => letter === q[ix]);
```

`every` subjects each item in an array to a test you provide. If every item passes, it returns `true`. If any fails, it returns `false`. Because you've sorted the letters in each array, those at the same index must be compared. No inner loop required!

A useful companion to `every` is `some`. As always [Mozilla](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String) is the best reference.

**Comment by Alex-Rafter** [comment]
Yes that's much cleaner, as well as solving the nested loop problem. Updated in the new version. Thanks! 👍
